proposals/dnd-list.md

```diff
@@ -0,0 +1,144 @@
+# Drag and Drop List
+
+## Usage
+
+### Basic usage
+
+```js
+import * as React from 'react';
+import {StatefulList} from 'baseui/dnd-list';
+
+export default () =>
+  <StatefulList
+    initialState={{items: ['Item 1', 'Item 2', 'Item 3']}}
+    onChange={({newState, oldIndex, newIndex}) => {}}
+  />
+```
+
+### Advanced usage
+
+```js
+import * as React from 'react';
+import {
+  StatefulList,
+  StyledList,
+  StyledItem
+} from 'baseui/dnd-list';
+import {styled} from 'baseui';
+
+CustomList = styled(StyledList, props => ({
+  border: '1px solid green'
+}));
+
+CustomItem = styled(StyledItem, props => ({
+  borderBottom: props.$dragged ? `1px solid red` : 'none'
+}));
+
+export default () =>
+  <StatefulList
+    initialState={{ items: ['Item 1', 'Item 2', 'Item 3']}}
+    overrides={{
+      List: CustomList,
+      Item: CustomItem
+    }}
+  >
+```
+
+## Exports
+
+* `StatefulList`
+* `List`
+* `StyledList`
+* `StyledItem`
+* `StyledDragHandle`
+* `StyledCloseHandle`
+* `STATE_CHANGE_TYPE`
+
+## `List` API
+
+* `items: Array<string>` - Required.
```

Comment:
What is a `string` item here? Should it be a `React.Node` or a component?

Reply:
It should be the item label: `Draggable List Item One`. I guess React.Node could be a better choice because localization libs usually use some React component to translate strings. But the expectation is that the output should be a plain text to keep the design nice. Wdyt?

<img width="1152" alt="screen shot 2018-11-29 at 2 47 00 pm" src="https://user-images.githubusercontent.com/1387913/50849958-5dd1d700-132d-11e9-9bc1-a5aece403215.png">

Reply:
Let’s do React.Node, so it provides more flexibility if there is some additional markup needed.

Reply:
Updated.